### PR TITLE
[WIP][xbmc][deps] My ongoing work to get Win64 done

### PR DESCRIPTION
**Pull request by Paxxi**
## Discussions can wait until it's in testing stage

For simplicity, The libs marked as building will be available in a forked version under https://github.com/Paxxi/ with any changes I've made to them to get them building

First milestone reached. All libs required to compile Kodi for win64 builds!!

haha, great success!
image

| Builds | Tested | Packaged | Lib |
| --- | --- | --- | --- |
| :white_check_mark: | :white_large_square: | :white_check_mark: | ffmpeg |
| :white_check_mark: | :white_large_square: | :white_check_mark: | libdvd |
| :white_large_square: | :white_large_square: | :white_large_square: | libbz2 (don't seem to be needed) |
| :white_check_mark: | :white_large_square: | :white_large_square: | crossguid |
| :white_check_mark: | :white_large_square: | :white_large_square: | curl |
| :white_check_mark: | :white_large_square: | :white_large_square: | dnssd |
| :white_large_square: | :white_large_square: | :white_large_square: | fontconfig (don't seem to be needed now) |
| :white_check_mark: | :white_large_square: | :white_large_square: | freetype |
| :white_large_square: | :white_large_square: | :white_large_square: | giflib (only for texturepacker) |
| :white_check_mark: | :white_large_square: | :white_large_square: | libass |
| :white_check_mark: | :white_large_square: | :white_large_square: | libbluray |
| :white_check_mark: | :white_large_square: | :white_large_square: | libcdio |
| :white_check_mark: | :white_large_square: | :white_large_square: | libcec |
| :white_check_mark: | :white_large_square: | :white_large_square: | expat |
| :white_check_mark: | :white_large_square: | :white_large_square: | fribidi |
| :white_check_mark: | :white_large_square: | :white_large_square: | iconv (builds hopefully fixed the weird pointer usage) |
| :white_large_square: | :white_large_square: | :white_large_square: | jpeg-turbo (only for texturepacker) |
| :white_check_mark: | :white_large_square: | :white_large_square: | libmicrohttpd |
| :white_check_mark: | :white_large_square: | :white_large_square: | libnfs |
| :white_check_mark: | :white_large_square: | :white_large_square: | libplist |
| :white_large_square: | :white_large_square: | :white_large_square: | libpng (only for texturepacker) |
| :white_large_square: | :white_large_square: | :white_large_square: | librtmp (can be removed afaik) |
| :white_check_mark: | :white_large_square: | :white_large_square: | libshairplay |
| :white_check_mark: | :white_large_square: | :white_large_square: | libssh |
| :white_check_mark: | :white_large_square: | :white_large_square: | libxml2 |
| :white_check_mark: | :white_large_square: | :white_large_square: | libxslt |
| :white_check_mark: | :white_large_square: | :white_large_square: | lzo |
| :white_check_mark: | :white_large_square: | :white_large_square: | libmysql |
| :white_check_mark: | :white_large_square: | :white_large_square: | openssl |
| :white_check_mark: | :white_large_square: | :white_large_square: | pcre |
| :white_large_square: | :white_large_square: | :white_large_square: | pillow |
| :white_check_mark: | :white_large_square: | :white_large_square: | python |
| :white_check_mark: | :white_large_square: | :white_large_square: | sqlite |
| :white_check_mark: | :white_large_square: | :white_large_square: | taglib |
| :white_check_mark: | :white_large_square: | :white_large_square: | tinyxml |
| :white_check_mark: | :white_large_square: | :white_large_square: | zlib |

### CODE REVIEW

**Comment by hudokkow** [comment]
ouch! Sounds like you have too much free time on your hands!

File: tools/buildsteps/win32/download-msys.bat [hunk 1]
```diff
@@ -58,7 +58,7 @@ cd %CUR_PATH%
 rem insert call to vsvars32.bat in msys.bat
 cd %MSYS_INSTALL_PATH%
 Move msys.bat msys.bat_dist
-ECHO CALL "%VS140COMNTOOLS%\..\..\VC\bin\vcvars32.bat">>msys.bat
+ECHO CALL "%VS140COMNTOOLS%\..\..\VC\amd64\vcvars64.bat">>msys.bat
```
Comment (afedchin) [inline tools/buildsteps/win32/download-msys.bat]: this file is obsolete.

File: tools/buildsteps/win32/make-mingwlibs.bat [hunk 2]
```diff
@@ -4,22 +4,22 @@ SETLOCAL
...
-call "%VS140COMNTOOLS%\..\..\VC\bin\vcvars32.bat" || exit /b 1
+call "%VS140COMNTOOLS%\..\..\VC\bin\amd64\vcvars64.bat" || exit /b 1
 
 SET PROMPTLEVEL=prompt
 SET BUILDMODE=clean
 SET opt=mintty
-SET build32=yes
-SET build64=no
+SET build32=no
+SET build64=yes
```
Comment (afedchin) [inline tools/buildsteps/win32/make-mingwlibs.bat]: please don't change default values, if you pass a build64 param then it should build x64 version of libs. also need to call to vcvarsXX.bat depending on build.

File: tools/windows/prepare-binary-addons-dev.bat [hunk 3]
```diff
@@ -18,7 +18,7 @@ FOR %%b IN (%*) DO (
 SETLOCAL DisableDelayedExpansion
 
 rem set Visual C++ build environment
-call "%VS120COMNTOOLS%..\..\VC\bin\vcvars32.bat"
+call "%VS120COMNTOOLS%..\..\VC\bin\amd64\vcvars64.bat"
```
Comment (afedchin) [inline tools/windows/prepare-binary-addons-dev.bat]: same as above

**Comment by Paxxi** [comment]
@afedchin this is just quick and dirty to test compile Kodi itself. However when this gets done I have no intention of supporting 32 bit builds anymore as it'll be a pain to keep all libs in shape for both platforms.

**Comment by fritsch** [comment]
Go go go :-) already > 50% done

**Comment by afedchin** [comment]
good progress, but I think we should keep supporting of 32 bit version. It makes no sense to install x64 windows for HTPC with 4GB mem.

**Comment by fritsch** [comment]
@afedchin: Think of CPU intrinsics. Especially SSEX or later can be enabled by default.

**Comment by afedchin** [comment]
@fritsch intrinsics aren't a reason to drop x86 support.

**Comment by Paxxi** [comment]
@afedchin the increased memory space is a common misconception about 64 bit. Yes it's valuable for some applications but not for us, the important thing for us are things like increased registers and register space and other improvements to the cpu  architecture

**Comment by afedchin** [comment]
It's about 8% of all gaming PCs (from steam stats http://store.steampowered.com/hwsurvey/?platform=pc) and I think this value is greater for HTPC

**Comment by Paxxi** [comment]
If unified depends were fully up and running for Windows it would be fairly easy to support both and this might still happen before this work is done. The current situation of building and packaging libs are painful enough as it is, doubling the work to maintain both x86 and x64 is not something I'm willing to do for 8% of PCs worldwide

File: xbmc/system.h [hunk 4]
```diff
@@ -94,6 +94,10 @@
  *****************/
 
 #if defined(TARGET_WINDOWS)
+<<<<<<< e2f22cf7d1868a0e8ba19c02772b270cc145bd32
+=======
+//#define HAS_DVD_DRIVE
+>>>>>>> Fixed some build issues
```
Comment (afedchin) [inline xbmc/system.h]: conflict artifact.

File: xbmc/storage/cdioSupport.cpp [hunk 5]
```diff
@@ -20,6 +20,7 @@
 
 #include "system.h"
 
+
```
Comment (afedchin) [inline xbmc/storage/cdioSupport.cpp]: unrelated change

**Comment by Rechi** [comment]
superseded by #12065
